### ISSUE TITLE
Assertion error when running Andersen pointer analysis

### ISSUE THREAD

**Issue by jo285317** [SVF-tools/SVF#179]
Tried running: wpa --ander --stat vmlinux.bc
Stderr:
wpa: /SVF/include/MemoryModel/MemModel.h:571: SymID SymbolTableInfo::getValSym(const Value*): Assertion `iter!=valSymMap.end() &&"value sym not found"' failed.

I obtained the LLVM bit code with wllvm and I'm using llvm-9.

Any suggestions would be appreciated

**Comment by yuleisui** [comment]
Would you be able to provide the stack trace and the LLVM value which is not found?

If the bc is large, you could also give us a link to download.

**Comment by jo285317** [comment]
There is no stack trace, this is the complete output:

```
nod@xps13:~/linux-5.4.4$ /home/nod/apps/SVF/Release-build/bin/wpa --ander --dump-callgraph vmlinux.bc
wpa: /home/nod/apps/SVF/include/MemoryModel/MemModel.h:571: SymID SymbolTableInfo::getValSym(const Value*): Assertion `iter!=valSymMap.end() &&"value sym not found"' failed.
Aborted (core dumped)
```
[Here is the bc, hosted on google drive](https://drive.google.com/file/d/1T1m88hJhuBY7_FnJ3jdVh1g1feJywjYl/view?usp=sharing)

**Comment by yuleisui** [comment]
I suspect the value `val` is not included in the SymMap when collecting symbols. Could you let us  know the LLVM value of `val`?

**Comment by jo285317** [comment]
Is this what you need?

<pre>(gdb) print *val
$3 = {VTy = <font color="#0000AA">0x5555563e17e0</font>, UseList = <font color="#0000AA">0x55555bbbdb20</font>, SubclassID = 3 &apos;\003&apos;, HasValueHandle = 0 &apos;\000&apos;, 
  SubclassOptionalData = 0 &apos;\000&apos;, SubclassData = 0, NumUserOperands = 1, IsUsedByMD = 0, HasName = 1, 
  HasHungOffUses = 0, HasDescriptor = 0, static MaxAlignmentExponent = 29, 
  static MaximumAlignment = 536870912}
</pre>

**Comment by yuleisui** [comment]
can you do "val->dump()"?

**Comment by jo285317** [comment]
tried to print val->dump() from gdb. i get: Cannot evaluate function -- may be inlined

**Comment by shiheyuan** [comment]
I met the same problem when I run `wpa --nander vmlinux.bc`. The same error log:

wpa: SVF/include/MemoryModel/MemModel.h:571: SymID SymbolTableInfo::getValSym(const Value*): Assertion `iter!=valSymMap.end() &&"value sym not found"' failed.

**Comment by shiheyuan** [comment]
> Tried running: wpa --ander --stat vmlinux.bc
> Stderr:
> wpa: /SVF/include/MemoryModel/MemModel.h:571: SymID SymbolTableInfo::getValSym(const Value*): Assertion `iter!=valSymMap.end() &&"value sym not found"' failed.
> 
> I obtained the LLVM bit code with wllvm and I'm using llvm-9.
> 
> Any suggestions would be appreciated

Hi @jo285317.  Do you solve this problem?
And can you give me some guidance on how to build kernel by llvm? 
By the way, I tried to use wllvm(llvm 9.0.0) to build Linux-5.4.6 source, and receive lots of logs like "Failed to find link section for the section XX" in the build procession.

Thanks.

**Comment by bsauce** [comment]
Is this problem solved? I have the same error too.

**Comment by yuleisui** [comment]
Yes, solved in the lastest SVF version. Please try and let me know.

**Comment by bsauce** [comment]
> Yes, solved in the lastest SVF version. Please try and let me know.

I installed the newst SVF, but there is another error when I analyzed vmlinux.

```
john@ubuntu:~/Desktop/test/bitcode$ wpa -indCallLimit=100000 -dump-callgraph -ander -svfg -dump-mssa ./vmlinux.bc > test.txt
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
Aborted (core dumped)
```

**Comment by yuleisui** [comment]
It looks to me that this is caused by running out of memory on your local machine. How about running Andersen's analysis alone `-ander`?

**Comment by bsauce** [comment]
> It looks to me that this is caused by running out of memory on your local machine. How about running Andersen's analysis alone `-ander`?

Yes, you are right. I only have 55G memory. How much memory does it need to analyze a kernel?
